mempool/clist_mempool.go

```diff
@@ -700,20 +772,18 @@ type recheck struct {
 	recheckFull   atomic.Bool     // whether rechecking TXs cannot be completed before a new block is decided
 }
 
-func newRecheck() *recheck {
-	return &recheck{
-		doneCh: make(chan struct{}, 1),
-	}
-}
-
 func (rc *recheck) init(first, last *clist.CElement) {
 	if !rc.done() {
 		panic("Having more than one rechecking process at a time is not possible.")
 	}
 	rc.cursor = first
 	rc.end = last
+	rc.doneCh = make(chan struct{})
```

Comment:
`doneCh` had previously capacity 1. Is this change intended?

Reply:
I think I understand: is it because we're now closing the channel, rather than sending `struct{}{}`?
If that is the case, please check [this comment](https://github.com/cometbft/cometbft/pull/3638/files?diff=split&w=1#r1711826972).